spec/controllers/saml_idp_controller_spec.rb

```diff
@@ -211,6 +211,29 @@
       end
     end
 
+    context 'service provider is valid' do
+      before do
```

Comment:
IIUC this project consistently tries to put all the context-setting actions in the `before` block. Every `context` should have a `before` to establish the described conditions. That's true for a single `it` or multiple `it` blocks.